internal/infrastructure/storage/db/postgres/utxo_repository.go

```diff
@@ -0,0 +1,906 @@
+package postgresdb
+
+import (
+	"context"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/vulpemventures/ocean/internal/core/domain"
+	"github.com/vulpemventures/ocean/internal/infrastructure/storage/db/postgres/sqlc/queries"
+	"sync"
+	"time"
+)
+
+const (
+	utxoSpent = iota
+	utxoConfirmed
+)
+
+type utxoRepositoryPg struct {
+	pgxPool          *pgxpool.Pool
+	querier          *queries.Queries
+	chLock           *sync.Mutex
+	chEvents         chan domain.UtxoEvent
+	externalChEvents chan domain.UtxoEvent
+}
+
+func NewUtxoRepositoryPgImpl(pgxPool *pgxpool.Pool) domain.UtxoRepository {
+	return &utxoRepositoryPg{
+		pgxPool:          pgxPool,
+		querier:          queries.New(pgxPool),
+		chLock:           &sync.Mutex{},
+		chEvents:         make(chan domain.UtxoEvent),
+		externalChEvents: make(chan domain.UtxoEvent),
+	}
+}
+
+func (u *utxoRepositoryPg) AddUtxos(
+	ctx context.Context,
+	utxos []*domain.Utxo,
+) (int, error) {
```

Comment:
I see there are a few `tx.Rollback` invocations in case of errors.
One thing we can do to avoid such code repetition is to give names to the returns values like:
```suggestion
) (num int, err error) {
```

This way we could defer a function like:
```
defer func(){
   if err != nil {
      tx.Rollback(ctx)
   }
}()
```

I would change all methods making use of db txs like this.

Suggestion: I would set the value of `num` to `-1` at the very beginning of this func and set it to `count` only at the very end of it, so the method always returns `-1, err` or `num, nil`

Reply:
Here the thing is that i was wondering if we should put all insertion inside same tx or not, i decided to go with option to insert correct ones and that failed dont impact correct ones, but maybe we should treat them all or nothing? @altafan

Reply:
All or nothing is the definition of db transaction. I think we should go for that approach, and roll it back in case of errors